libs/core/langchain_core/messages/utils.py

```diff
@@ -466,6 +472,27 @@ def filter_messages(
         else:
             pass
 
+        if exclude_tool_call_ids is True and (
+            (isinstance(msg, AIMessage) and msg.tool_calls)
+            or isinstance(msg, ToolMessage)
+        ):
+            continue
+        if isinstance(exclude_tool_call_ids, list):
+            if isinstance(msg, AIMessage) and msg.tool_calls:
+                tool_calls = [
+                    tool_call
+                    for tool_call in msg.tool_calls
+                    if tool_call["id"] not in exclude_tool_call_ids
+                ]
+                if not tool_calls:
+                    continue
+                msg = msg.model_copy(update={"tool_calls": tool_calls})
```

Comment:
this needs to be a bit more involved for Anthropic models - you would need to update tool calls in the content list as well